### PR TITLE
FSM integration

### PR DESCRIPTION
**Pull request by Lucien950**
### Summary
- Make FSM **comp ready**

### Changelist
- Change IOC settings to accept digital signals
- Correct Analogue and Digital sensor channels
- Correct transition functions between voltages and real world values

### Testing Done
- Board testing
- (hopefully sensors testing on car later)

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/FSM/Src/Io/Io_Brake.c [hunk 1]
```diff
@@ -3,8 +3,8 @@
 #include "Io_Adc.h"
 
 // TODO set these values
```
Comment (gtaharaedmonds) [inline boards/FSM/Src/Io/Io_Brake.c:5]: Can we remove this now?

File: boards/FSM/Src/Io/Io_Coolant.c [hunk 2]
```diff
@@ -37,46 +38,50 @@ void Io_FlowMeter_CheckIfFlowMeterIsActive(void)
...
-    // TODO calculate the temperature
-    return 1;
+    return voltageToTemperature(temperature_voltage);
 }
 
+float voltageToPressure(float v)
+{
+    // TODO calculate the pressure
+    return v;
+}
```
Comment (gtaharaedmonds) [inline boards/FSM/Src/Io/Io_Coolant.c:69]: What's the plan for these? Another PR
(same with temperature above)

File: boards/FSM/Src/Io/Io_SoftwareWatchdog.c [hunk 3]
```diff
@@ -13,7 +13,7 @@ void Io_HardwareWatchdog_Refresh(void)
 
 void Io_SoftwareWatchdog_TimeoutCallback(SoftwareWatchdogHandle_t watchdog)
 {
-    BREAK_IF_DEBUGGER_CONNECTED();
+    //    BREAK_IF_DEBUGGER_CONNECTED();
```
Comment (gtaharaedmonds) [inline boards/FSM/Src/Io/Io_SoftwareWatchdog.c:16]: Forgot about this line, just asked Kiron what he thinks of this change

File: boards/FSM/Src/Io/Io_Brake.c (hunk 1 above)
Comment (Lucien950) [reply boards/FSM/Src/Io/Io_Brake.c:5]: These are still guesses, especially for the pedal

File: boards/FSM/Src/Io/Io_Coolant.c (hunk 2 above)
Comment (Lucien950) [reply boards/FSM/Src/Io/Io_Coolant.c:69]: Joe said we don't know what sensors we are using yet. Once that gets decided, I will implement the transfer functions

File: boards/FSM/Src/Io/Io_SoftwareWatchdog.c (hunk 3 above)
Comment (Lucien950) [reply boards/FSM/Src/Io/Io_SoftwareWatchdog.c:16]: Removed.

File: boards/FSM/Src/Io/Io_SoftwareWatchdog.c [hunk 4]
```diff
@@ -13,8 +13,6 @@ void Io_HardwareWatchdog_Refresh(void)
 
 void Io_SoftwareWatchdog_TimeoutCallback(SoftwareWatchdogHandle_t watchdog)
 {
-    BREAK_IF_DEBUGGER_CONNECTED();
-
```
Comment (gtaharaedmonds) [inline boards/FSM/Src/Io/Io_SoftwareWatchdog.c:17]: Can you add this back for now? Want to do more testing to determine if this is really a problem before we commit to removing it.

**Review by gtaharaedmonds — APPROVED** [review]
LGTM 
Can you add back the break if debugger connected and then go ahead and merge

**Review by gtaharaedmonds — CHANGES_REQUESTED** [review]
Actually wait can you fix the tests first? Check not successful currently

**Comment by Lucien950** [comment]
passing! transfer functions still not implemented tho.

**Review by gtaharaedmonds — APPROVED** [review]

**Pull request merged**